src/libos/crates/host-socket/src/common/sockopt.rs

```diff
@@ -0,0 +1,280 @@
+            level, optname, optval, optlen,
+        ))),
+    })
+}
+
+pub fn new_setsockopt_cmd(
+    level: c_int,
+    optname: c_int,
+    optval: *const c_void,
+    optlen: socklen_t,
+) -> Result<Box<dyn IoctlCmd>> {
+    if level != SOL_SOCKET {
+        return Ok(Box::new(SetSockOptRawCmd::new(SetSockOptRawInput::new(
+            level, optname, optval, optlen,
+        ))));
+    }
+
+    let opt =
+        SockOptName::try_from(optname).map_err(|_| errno!(ENOPROTOOPT, "Not a valid optname"))?;
+    Ok(match opt {
+        SockOptName::SO_ACCEPTCONN
+        | SockOptName::SO_DOMAIN
+        | SockOptName::SO_PEERNAME
+        | SockOptName::SO_TYPE
+        | SockOptName::SO_ERROR
+        | SockOptName::SO_PEERCRED
+        | SockOptName::SO_SNDLOWAT
+        | SockOptName::SO_PEERSEC
+        | SockOptName::SO_PROTOCOL
+        | SockOptName::SO_MEMINFO
+        | SockOptName::SO_INCOMING_NAPI_ID
+        | SockOptName::SO_COOKIE
+        | SockOptName::SO_PEERGROUPS => return_errno!(ENOPROTOOPT, "it's a read-only option"),
+        _ => Box::new(SetSockOptRawCmd::new(SetSockOptRawInput::new(
+            level, optname, optval, optlen,
+        ))),
+    })
+}
+
+#[derive(Debug)]
+pub struct GetSockOptRawInput {
+    level: c_int,
+    optname: c_int,
+    optval: *mut c_void,
+    optlen: *mut socklen_t,
+}
+
+impl GetSockOptRawInput {
+    pub fn new(level: c_int, optname: c_int, optval: *mut c_void, optlen: *mut socklen_t) -> Self {
+        Self {
+            level,
+            optname,
+            optval,
+            optlen,
+        }
+    }
+}
+
+impl_ioctl_cmd! {
+    pub struct GetSockOptRawCmd<Input=GetSockOptRawInput, Output=()> {}
+}
+
+impl GetSockOptRawCmd {
+    pub fn getsockopt(&mut self, fd: HostFd) -> Result<()> {
+        let input = self.input();
+        getsockopt_by_host(
+            fd as _,
+            input.level,
+            input.optname,
+            input.optval,
+            input.optlen,
+        )?;
+        self.set_output(());
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+pub struct SetSockOptRawInput {
+    level: c_int,
+    optname: c_int,
+    optval: *const c_void,
+    optlen: socklen_t,
+}
+
+impl SetSockOptRawInput {
+    pub fn new(level: c_int, optname: c_int, optval: *const c_void, optlen: socklen_t) -> Self {
+        Self {
+            level,
+            optname,
+            optval,
+            optlen,
+        }
+    }
+}
+
+impl_ioctl_cmd! {
+    pub struct SetSockOptRawCmd<Input=SetSockOptRawInput, Output=()> {}
+}
+
+impl SetSockOptRawCmd {
+    pub fn setsockopt(&mut self, fd: HostFd) -> Result<()> {
+        let input = self.input();
+        setsockopt_by_host(
+            fd as _,
+            input.level,
+            input.optname,
+            input.optval,
+            input.optlen,
+        )?;
+        self.set_output(());
+        Ok(())
+    }
+}
+
+impl_ioctl_cmd! {
+    pub struct SoAcceptConnCmd<Input=(), Output=i32> {}
```

Comment:
I suggest replace the `So` prefix (it basically contains zero info as the type belongs to a module named `sockopt`) with `Get` (it helps to differentiate between get or set commands) in the four commands of `SoAcceptConnCmd`, `SoDomainCmd`, `SoPeerNameCmd`, and `SoTypeCmd`.